### DISCUSSION TITLE
Separate groups for minor and patch packages

### DISCUSSION THREAD

**Discussion by mbmx10** [renovatebot/renovate#22832] [Request Help] [answered]
### How are you running Renovate?

Self hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.110.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Say we use dependency `@types/express` in version 1.16.0, for which 1.16.1 and 1.17.17 is published. We would like to see two grouped MRs, one for patches of `@types/express` and other configured dependencies and one for minor updates.

### What you tried so far.

[Edited] This is just an excerpt from a more complex configuration and might not include the root cause for the observation.

```
    {
      "additionalBranchPrefix": "backend/",
      "matchPaths": ["backend/**"],
      "matchUpdateTypes": ["minor"],
      "matchPackageNames": [
        "@types/express",
        "other-depencencies"
      ],
      "groupName": "minor upgrades notification"
    },
    {
      "additionalBranchPrefix": "backend/",
      "matchPaths": ["backend/**"],
      "matchUpdateTypes": ["patch"],
      "matchPackageNames": [
        "@types/express",
        "other-depencencies"
      ],
      "groupName": "patches notification"
    },
```

The result is a MR for the minor update, but no MR for the patch. (the patch was published four years ago in this specific case, if it matters). The new patch version 1.16.1 can not be found in the logs.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
checkout `separateMinorPatch ` config option

**Reply by mbmx10** [reply]
Thanks for the response. It took me a while to get back to this.

In my initial configuration example `"additionalBranchPrefix": "backend/",` was missing, which I identified now as a prerequisite for reproduction. `separateMinorPatch` did the job, after changing the values of `additionalBranchPrefix` to be different.

```
    {
      "additionalBranchPrefix": "minor/",
      "matchPaths": ["backend/**"],
      "matchUpdateTypes": ["minor"],
      "matchPackageNames": [
        "@types/express",
        "other-depencencies"
      ],
      "groupName": "minor upgrades notification"
    },
    {
      "additionalBranchPrefix": "patch/",
      "matchPaths": ["backend/**"],
      "matchUpdateTypes": ["patch"],
      "matchPackageNames": [
        "@types/express",
        "other-depencencies"
      ],
      "groupName": "patches notification"
    },
```

**Reply by viceice** [reply]
you don't need `additionalBranchPrefix` with `separateMinorPatch`, renovate will automatically add a suffix to branch name.

**Reply by mbmx10** [reply]
This is not what I wanted to express.

To be more clear: Having `additionalBranchPrefix` with the same value and `separateMinorPatch` did not give me the wanted end result. Once I changed `additionalBranchPrefix` to have different values, I got the wanted end result.

**Reply by viceice** [reply]
`separateMinorPatch=true` should give you two separate branches and two PR's. if this isn't the case please create a minimal reproduction, so we can check what's going wrong.

**Reply by mbmx10** [reply]
I tried to reproduce the combination from scratch and the result were two separate branches and two PRs. Unfortunately, the current configuration in our mono repo is too big to reduce to a minimal example for reproduction in a timely manner.
Since with my latest changes, the end result is as expected, I'd leave it with that.

Thanks again!
